src/update.c

```diff
@@ -208,9 +208,16 @@ static int main_update()
 	int current_version = -1, server_version = -1;
 	int mix_current_version = -1, mix_server_version = -1;
 	struct manifest *current_manifest = NULL, *server_manifest = NULL;
+	struct list *current_manifests = NULL;
+	struct list *server_manifests = NULL;
+	struct list *consolidated_manifests = NULL;
+	struct list *current_manifest_updates = NULL;
+	struct list *server_manifest_updates = NULL;
 	struct list *updates = NULL;
 	struct list *current_subs = NULL;
-	struct list *latest_subs = NULL;
+	struct list *initial_subs = NULL;
+	struct list *update_subs = NULL;
+	struct list *nonupdate_subs = NULL;
```

Comment:
It looks like the nonupdate_subs list is populated and freed, but never used for anything. Is there or will there be value for this?

Reply:
@jwakre the only reason that list exists is to make sure all the subscriptions are freed. 

Initially _current_subs_ has all the current subscribed bundles, then some of those subscriptions are copied to the _initial_subs_ , and _update_subs_ lists, then included bundles may be added to the _update_subs_ list. So some subscriptions end up being shared in many lists and others only appear in one list. When cleaning's up, we know all the subscriptions are in two lists without being duplicated: __update_subs_ and _non_update_subs_ . So if we want to free all the subscriptions we can just free them through these two lists.